frontend/simple/js/state.js

```diff
@@ -154,6 +199,18 @@ const actions = {
     commit('markMessageAsRead', hash)
     debouncedSave(dispatch)
   },
+  addUser ({dispatch, commit}, data) {
+    commit('addUser', data)
+    debouncedSave(dispatch)
+  },
+  removeContract ({dispatch, commit}, contractId) {
+    commit('removeContract', contractId)
+    debouncedSave(dispatch)
+  },
+  pending ({dispatch, commit}, contractId) {
+    commit('pending', contractId)
+    debouncedSave(dispatch)
+  },
```

Comment:
These actions exist only to call the debouncedSave function. This is code smell and looks like a DRY violation.

Let's instead do this:

- Call `store.subscribe` next to the debouncedSave definition, and simply call debouncedSave every time a commit is made.
- Delete calls to `debouncedSave` everywhere else.
- Delete these actions and revert (and replace as necessary) all `.dispatch`'s to `.commit`'s.